airflow/models/dagbag.py

```diff
@@ -107,28 +112,51 @@ def size(self):
     def dag_ids(self):
         return self.dags.keys()
 
-    def get_dag(self, dag_id):
+    def get_dag(self, dag_id, from_file_only=False):
         """
         Gets the DAG out of the dictionary, and refreshes it if expired
+
+        :param dag_id: DAG Id
+        :type dag_id: str
+        :param from_file_only: returns a DAG loaded from file.
+        :type from_file_only: bool
         """
         from airflow.models.dag import DagModel  # Avoid circular import
 
+        # Only read DAGs from DB if this dagbag is dagcached_enabled.
+        # from_file_only is an exception, currently it is for renderring templates
+        # in UI only. Because functions are gone in serialized DAGs, DAGs must be
+        # imported from files.
+        # FIXME: this exception should be removed in future, then webserver can be
+        # decoupled from DAG files.
+        if self.dagcached_enabled and not from_file_only:
+            print(self.dags)
+            return self.dags.get(dag_id)
+
         # If asking for a known subdag, we want to refresh the parent
+        dag = None
         root_dag_id = dag_id
         if dag_id in self.dags:
             dag = self.dags[dag_id]
             if dag.is_subdag:
                 root_dag_id = dag.parent_dag.dag_id
 
+        # Needs to load from file for a dagcached_enabled dagbag.
+        enforce_from_file = False
+        if self.dagcached_enabled and dag is not None:
+            from airflow.dag.serialization.serialized_dag import SerializedDAG
+            enforce_from_file = isinstance(dag, SerializedDAG)
+            self.log.info("Processing DAG file to render template.")
```

Comment:
This log shouldn't be here (but in the caller) -- all we know in this function is that we've been told to go to disk.

Reply:
Done